modules/economy-system/commands/shop.js

```diff
@@ -1,92 +1,42 @@
-const {createShopItem, editBalance, createShopMsg, deleteShopItem, createleaderboard} = require('../economy-system');
-const {embedType} = require('../../../src/functions/helpers');
+const {createShopItem, createShopMsg, deleteShopItem, shopMsg, buyShopItem} = require('../economy-system');
 const {localize} = require('../../../src/functions/localize');
 
+/**
+ * @param {*} interaction Interaction
+ * @returns {boolean} Result
+ */
+async function checkPerms(interaction) {
+    const result = interaction.client.configurations['economy-system']['config']['shopManagers'].includes(interaction.user.id) || interaction.client.config['botOperators'].includes(interaction.user.id);
+    if (!result) {
+        await interaction.reply({
+            content: interaction.client.strings['not_enough_permissions'],
+            ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
+        });
+    }
+    return result;
+}
+
 module.exports.subcommands = {
-    'add': async function (interaction) {
-        if (!interaction.client.configurations['economy-system']['config']['shopManagers'].includes(interaction.user.id) && !interaction.client.config['botOperators'].includes(interaction.user.id)) {
-            return await interaction.reply({
-                content: interaction.client.strings['not_enough_permissions'],
-                ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
-            });
-        }
-        const item = await interaction.options.get('item');
-        const price = await interaction.options.getInteger('price');
-        const role = await interaction.options.getRole('role', true);
-        await createShopItem(item['value'], price, role.id, interaction.client);
-        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['itemCreate'], {
-            '%item%': item['value'],
-            '%price%': price,
-            '%role%': role.name
-        }, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        interaction.client.logger.info(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: interaction.user.tag,
-            i: item['value']
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: interaction.user.tag,
-            i: item['value']
-        }));
+    'add': async function (interaction) { // ToDo: Update Command
+        if (!checkPerms(interaction)) return;
+        await createShopItem(interaction);
+        shopMsg(interaction.client);
```

Comment:
(so there is a todo left here…)
Also: For better UX, I would suggest checking if the bot's highest role is above the supplied role